cmd/yggd/main.go

```diff
@@ -236,6 +242,17 @@ func main() {
 			return cli.Exit(fmt.Errorf("unsupported transport protocol: %v", config.DefaultConfig.Protocol), 1)
 		}
 		client := NewClient(dispatcher, transporter)
+
+		enableMessageJournal := config.DefaultConfig.EnableMessageJournal
+		if enableMessageJournal {
+			journal, err := messagejournal.New(constants.ConfigDir, "messagejournal.db")
```

Comment:
Runtime data like this should not be written to the config directory; it belongs
in either `/var` or `/run`. If the configuration option is a filepath, you can
avoid this entirely and rely on the configuration to declare where to write the
message journal database file.